src/map/battle.c

```diff
@@ -5106,15 +5108,18 @@ static struct Damage battle_calc_weapon_attack(struct block_list *src, struct bl
 				break;
 		}
 
-		if( sd ) {
+		if (sd) {
```

Comment:
In every line that you touched where sd or other struct pointers are used please add != NULL.
One shall not mistreat non-boolean values as boolean values.